R/importTaxpasta.R

```diff
@@ -0,0 +1,116 @@
+#' Import taxpasta-specific BIOM results to `TreeSummarizedExperiment`
+#'
+#' @param filename A character vector with one element that denotes the file
+#'   path to a BIOM file.
+#'
+#' @return A [TreeSummarizedExperiment::TreeSummarizedExperiment()] object.
+#' @export
+#'
+#' @examples
+#' tse <- importTaxpasta("profiles.biom")
+#'
+#' @importFrom rhdf5 h5read
+#' @importFrom SummarizedExperiment rowData
+#' @importFrom SingleCellExperiment altExps
+importTaxpasta <- function(filename) {
+    # Validate the input.
+    if (!.is_non_empty_string(filename)) {
+        stop("'filename' must be a single character value.", call. = FALSE)
+    }
+    if (!file.exists(filename)) {
+        stop(filename, " not found.", call. = FALSE)
+    }
+
+    # We read our own HDF5 array to later be able to read observation group
+    # metadata, which [biomformat::read_biom()] currently doesn't do.
+    raw <- h5read(filename, "/", read.attributes = TRUE)
+    biom <- .createBiom(raw)
+
+    # Without taxonomic information, we return a simple TreeSE.
+    if (is.null(raw$observation$`group-metadata`$ranks)) {
+        warning("The BIOM file does not contain taxonomy information; unable to generate a taxonomic tree.")
+        return(makeTreeSEFromBiom(biom))
```

Comment:
importBIOM is the new name of the function

Reply:
I thought that `importBIOM` is for handling a file input, here `biom` is already a loaded BIOM object. At least, that's what the documentation suggests.

Reply:
You are correct. Sorry for misinformation